Swig can't build bindings if C++ was not installed in the superbuild

If  `CMAKE_INSTALL_PREFIX` is not set to `/robotology-superbuild/build/install` the swig interface compilation fails with:
```
[ 33%] Swig compile tcpi-python-types.i for python
/robotology-superbuild/build/install/share/yarp/bindings/yarp.i:437: Warning 206: Unexpected tokens after #endif directive.
/home/ecub/thrift-cpp-python-interop/python_library/tcpi-python-types.i:23: Error: Unable to find 'ThriftCppPythonInterop/Service.h'
make[2]: *** [CMakeFiles/tcpi_swig_compilation.dir/build.make:64: CMakeFiles/tcpi.dir/tcpi-python-typesPYTHON.stamp] Error 1
make[2]: *** Deleting file 'CMakeFiles/tcpi.dir/tcpi-python-typesPYTHON.stamp'
make[1]: *** [CMakeFiles/Makefile2:105: CMakeFiles/tcpi_swig_compilation.dir/all] Error 2
make: *** [Makefile:130: all] Error 2
```

This happens when `CMAKE_INSTALL_PREFIX` is set to an arbitrary location but even when it is not set and it defaults to `/usr/local`

--- a/python_library/CMakeLists.txt
+++ b/python_library/CMakeLists.txt
@@ -45,7 +45,7 @@ swig_add_library(${LIBRARY_TARGET_NAME}
                  SOURCES ${SWIG_BINDINGS_SOURCE_FILE})
 
 # Link libraries
-swig_link_libraries(${LIBRARY_TARGET_NAME} ThriftCppPythonInterop::ThriftCppPythonInterop Python3::Python YARP::YARP_os YARP::YARP_sig)
+target_link_libraries(${LIBRARY_TARGET_NAME} ThriftCppPythonInterop::ThriftCppPythonInterop Python3::Python YARP::YARP_os YARP::YARP_sig)
 
 # Include directories
 target_include_directories(${SWIG_MODULE_${LIBRARY_TARGET_NAME}_REAL_NAME} BEFORE PRIVATE ${Python3_INCLUDE_DIRS})

--- a/python_library/CMakeLists.txt
+++ b/python_library/CMakeLists.txt
@@ -43,6 +43,7 @@ set(CMAKE_SWIG_FLAGS "-py3;-O;-threads")
 swig_add_library(${LIBRARY_TARGET_NAME}
                  LANGUAGE python
                  SOURCES ${SWIG_BINDINGS_SOURCE_FILE})
+set_property(TARGET ${LIBRARY_TARGET_NAME} PROPERTY SWIG_USE_TARGET_INCLUDE_DIRECTORIES TRUE)
 
 # Link libraries
 target_link_libraries(${LIBRARY_TARGET_NAME} ThriftCppPythonInterop::ThriftCppPythonInterop Python3::Python YARP::YARP_os YARP::YARP_sig)

